### PR TITLE
Add links to room details in calendar page

### LINKED ISSUE #3531: Add links to room details from Calendar page

### PR DESCRIPTION
**Pull request by mvidalgarcia**
Closes #3531

### CODE REVIEW

File: indico/modules/rb_new/client/js/components/TimelineContent.jsx [hunk 1]
```diff
@@ -152,7 +156,10 @@ function TimelineRowLabel({label, availability, longLabel}) {
     const roomLabel = (
         <span>
             {availability && <Icon name="circle" size="tiny" color={color} styleName="dot" />}
-            <span>{label}</span>
+            {onClickLabel
+                ? <a style={{cursor: 'pointer'}} onClick={onClickLabel}>{label}</a>
```
Comment (pferreir) [inline indico/modules/rb_new/client/js/components/TimelineContent.jsx]: Is this the only way to make it look like a link?

Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/components/TimelineContent.jsx]: I think the only other option would be adding a `href` attribute (which would then require you to preventDefault the regular click event)... or maybe we could set this cursor style on all `a` elements in rb-new? We don't use non-link anchors anyway.

Comment (mvidalgarcia) [reply indico/modules/rb_new/client/js/components/TimelineContent.jsx]: I got it from https://github.com/indico/indico/blob/master/indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx#L308 but I agree on setting this cursor style on all `a` elements in rb-new.

File: indico/modules/rb_new/client/js/components/TimelineContent.jsx [hunk 2]
```diff
@@ -69,7 +73,8 @@ export default class TimelineContent extends React.Component {
             <div styleName="timeline-row" key={`element-${key}`}>
                 <TimelineRowLabel label={label}
                                   availability={conflictIndicator ? (hasConflicts ? 'conflict' : 'available') : null}
-                                  longLabel={longLabel} />
+                                  longLabel={longLabel}
+                                  onClickLabel={onClickLabel && (() => onClickLabel(key))} />
```
[the remark this answers is not in the document]
Comment (mvidalgarcia) [reply indico/modules/rb_new/client/js/components/TimelineContent.jsx]: Didn't notice I could use `room.id` :man_facepalming:

[the remark this answers is not in the document]
Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/components/TimelineContent.jsx]: Isn't it a booking ID?

[the remark this answers is not in the document]
Comment (mvidalgarcia) [reply indico/modules/rb_new/client/js/components/TimelineContent.jsx]: No, it's the id of the room whose info is printed on each timeline row.

[the remark this answers is not in the document]
Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/components/TimelineContent.jsx]: Ah it's room details. I somehow thought it's booking details.

**Comment by ThiefMaster** [comment]
The dimmer/loader from the room preloader doesn't seem to be working while loading the room details on the calendar page.

**Comment by mvidalgarcia** [comment]
Done.

File: indico/modules/rb_new/client/js/components/TimelineContent.jsx [hunk 3]
```diff
@@ -152,7 +159,10 @@ function TimelineRowLabel({label, availability, longLabel}) {
     const roomLabel = (
         <span>
             {availability && <Icon name="circle" size="tiny" color={color} styleName="dot" />}
-            <span>{label}</span>
+            {onClickLabel
```
Comment (mic4ael) [inline indico/modules/rb_new/client/js/components/TimelineContent.jsx]: wouldn't it fit in a single line?

**Pull request merged**